lib/workers/global/config/parse/host-rules-from-env.ts

```diff
@@ -1,15 +1,41 @@
 import { logger } from '../../../../logger';
 import { getDatasourceList } from '../../../../modules/datasource';
+import { getPlatformList } from '../../../../modules/platform';
 import type { HostRule } from '../../../../types';
 
 type AuthField = 'token' | 'username' | 'password';
 
+type HttpsAuthField =
+  | 'httpscertificate'
+  | 'httpsprivatekey'
+  | 'httpscertificateauthority';
+
 function isAuthField(x: unknown): x is AuthField {
   return x === 'token' || x === 'username' || x === 'password';
 }
 
+function isHttpsAuthField(x: unknown): x is HttpsAuthField {
+  return (
+    x === 'httpscertificate' ||
+    x === 'httpsprivatekey' ||
+    x === 'httpscertificateauthority'
+  );
+}
+
+function restoreHttpsAuthField(x: HttpsAuthField): string {
+  switch (x) {
+    case 'httpsprivatekey':
+      return 'httpsPrivateKey';
+    case 'httpscertificate':
+      return 'httpsCertificate';
+    case 'httpscertificateauthority':
+      return 'httpsCertificateAuthority';
+  }
+}
+
 export function hostRulesFromEnv(env: NodeJS.ProcessEnv): HostRule[] {
   const datasources = new Set(getDatasourceList());
+  const platforms = new Set(getPlatformList());
```

Comment:
bad idea. this gives transitive circular reference which can cause runtime issues. why do you need the host types of all platforms?

Reply:
I was just following the patterns. I don't need other host types, I've made it just for github for now.